Code/GraphMol/DetermineBonds/catch_tests.cpp

```diff
@@ -225,4 +225,118 @@ H     -1.187721   -0.949657    0.563607
     CHECK(m->getAtomWithIdx(1)->getChiralTag() ==
           Atom::ChiralType::CHI_TETRAHEDRAL_CCW);
   }
+}
+
+TEST_CASE("Github #5902: memory-hungry iterating") {
```

Comment:
a question since I haven't run this myself: do you have a sense of how long it takes to run this test?
I'm concerned about adding something which takes a long time to run to the standard test suite.

Reply:
For this case the algorithm finds a usable set of valences almost immediately, maybe even even the first entry in the list, which is why it was a shame to spend so much memory making the list of lists.  If the fix here works this returns very quickly.

If the fix it is testing doesn't work, the failure mode would be consuming a large amount of memory and....well on my machine it crashes, but I guess if a test machine had enough memory the test would still work.  So maybe this is a bad test?  I can remove it, I can't think of another test though.

I think I addressed the comments - let me know if I missed anything

Reply:
My primary concern was adding a test which makes the CI builds take forever. If this runs quickly then I'm not worried about it.